test/test_text_generator.py

```diff
@@ -1,102 +1,118 @@
+import inspect
 import unittest
 from unittest.mock import MagicMock
 
 import src.text_generator as text_gen
+from src.csvDatabase import CSVDatabase
+from src.database_context import DatabaseContext
+from src.invoice_generator import Invoice
 from src.text_generator import Database
+from src.user_dto import UserDto
 
-
-class InvoiceMock:
-    products = []
-    number = 0
-    payment = 'VISA/Mastercard - 1234'
-    date = ''
-    price = 0
+import utils.data_set as data_set
 
 
 class TestReplaceKeywords(unittest.TestCase):
+    def setUp(self) -> None:
+        self.context = DatabaseContext("db")
+        self.db = Database(self.context)
+
+    def tearDown(self) -> None:
+        self.context.destroy()
     def test_greeting(self):
-        database = Database()
-        database.get_user_sex = MagicMock(return_value='M')
-        database.get_user_surname = MagicMock(return_value='Wierzba')
+        self.db.get_user_sex = MagicMock(return_value='M')
+        self.db.get_user_surname = MagicMock(return_value='Wierzba')
 
-        assert text_gen.replace_keywords('{$greeting}', 1, database) == 'Szanowny Panie Wierzba'
+        assert text_gen.replace_keywords('{$greeting}', 1, self.db) == 'Szanowny Panie Wierzba'
```

Comment:
`text_gen.replace_keywords('{$greeting}', 1, self.db)`
`Expected type 'str', got 'int' instead---^`

Same problem happens in other test

Reply:
changed